src/OrchardCore/OrchardCore.Abstractions/Modules/IAsyncStartup.cs

```diff
@@ -0,0 +1,20 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+
+namespace OrchardCore.Modules
+{
+    /// <summary>
+    /// An implementation of this interface allows to configure asynchronously the tenant pipeline.
+    /// </summary>
+    public interface IAsyncStartup
+    {
+        /// <summary>
+        /// This method gets called before any <see cref="IStartup.Configure"/> and can collaborate to
+        /// build the tenant pipeline, but is not intended to configure the route/endpoint middleware.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="serviceProvider"></param>
+        ValueTask ConfigureAsync(IApplicationBuilder builder, IServiceProvider serviceProvider);
```

Comment:
Why you are using `ValueTask` instead of `Task`?

Reply:
As I remember if it executes asynchronously, which will be the case if the implementation is overridden, `Task` may be better.

But most of the time it will not be overridden, so the base class implementation which just returns a ValueTask will be executed synchronously, in that case ValueTask is lighter.